src/Disks/ObjectStorages/AzureBlobStorage/AzureObjectStorage.cpp

```diff
@@ -127,25 +127,22 @@ bool AzureObjectStorage::exists(const StoredObject & object) const
 {
     auto client_ptr = client.get();
 
-    /// What a shame, no Exists method...
-    Azure::Storage::Blobs::ListBlobsOptions options;
-    options.Prefix = object.remote_path;
-    options.PageSizeHint = 1;
-
-    ProfileEvents::increment(ProfileEvents::AzureListObjects);
+    ProfileEvents::increment(ProfileEvents::AzureGetProperties);
     if (client_ptr->GetClickhouseOptions().IsClientForDisk)
-        ProfileEvents::increment(ProfileEvents::DiskAzureListObjects);
-
-    auto blobs_list_response = client_ptr->ListBlobs(options);
-    auto blobs_list = blobs_list_response.Blobs;
+        ProfileEvents::increment(ProfileEvents::DiskAzureGetProperties);
 
-    for (const auto & blob : blobs_list)
+    try
     {
-        if (object.remote_path == blob.Name)
-            return true;
+        auto blob_client = client_ptr->GetBlobClient(object.remote_path);
+        blob_client.GetProperties();
+        return true;
+    }
+    catch (const Azure::Storage::StorageException & e)
+    {
+        if (e.StatusCode == Azure::Core::Http::HttpStatusCode::NotFound)
+            return false;
+        throw;
```

Comment:
Actually I don't know how to test negative case. Unit test requires mock of azure blob storage which is tricky.

Reply:
maybe you can add a failpoint?

Reply:
You could try to use Asure table function and write some files which already present. It fails unless you set `_truncate_on_insert` or `_create_new_file_on_insert` settings. And vice versa if there is no file exists it is added.